core-web/libs/portlets/edit-ema/portlet/src/lib/dot-ema-shell/dot-ema-shell.component.ts

```diff
@@ -313,11 +266,56 @@ export class DotEmaShellComponent implements OnInit, OnDestroy {
                     newLanguage: language.id
                 });
             },
-            reject: () => {
-                this.navigate({
-                    language_id: 1
-                });
-            }
+            reject: () => this.#goBackToCurrentLanguage()
         });
     }
+
+    /**
+     * Get the query params from the Router
+     *
+     * @return {*}  {DotPageApiParams}
+     * @memberof DotEmaShellComponent
+     */
+    #getPageParams(): DotPageApiParams {
+        const { queryParams, data } = this.#activatedRoute.snapshot;
+        const uveConfig = data?.uveConfig;
+        const allowedDevURLs = uveConfig?.options?.allowedDevURLs;
+
+        // Clone queryParams to avoid mutation errors
+        const params = filterPageParams(queryParams);
+        const validHost = checkClientHostAccess(params.clientHost, allowedDevURLs);
+
+        if (!validHost) {
+            delete params.clientHost;
+        }
+
+        if (uveConfig.url && !validHost) {
+            params.clientHost = uveConfig.url;
+        }
+
+        return params;
+    }
+
+    /**
+     * Update the location with the new query params
+     *
+     * Note: This method does not trigger a navigation event
+     *
+     * @param {Params} queryParams
+     * @memberof DotEmaShellComponent
+     */
+    #updateLocation(queryParams: Params = {}): void {
+        const urlTree = this.#router.createUrlTree([], { queryParams });
+
+        this.#location.replaceState(urlTree.toString());
```

Comment:
Does this trigger a re-render?

Reply:
No! It just replace the URL with the new qp

Reply:
I upload some videos of how it works in traditional and headless